File: app/src/main/java/com/hfad/avc/ui/main_avtivity/MainPresenter.java
package com.hfad.avc.ui.main_avtivity;

import com.github.terrakok.cicerone.Router;
import com.hfad.avc.cicerone.Screens;
import com.hfad.avc.dagger.ComponentManager;

import javax.inject.Inject;

import moxy.InjectViewState;
import moxy.MvpPresenter;


@InjectViewState
public class MainPresenter extends MvpPresenter<IMainViewModel> {

   @Inject
   Router router;

   public MainPresenter() {
      ComponentManager.Companion.getInstance().appComponent.inject(this);
   }

   @Override
   protected void onFirstViewAttach() {
      super.onFirstViewAttach();
      this.router.navigateTo(Screens.main());
   }
}

File: app/src/main/java/com/hfad/avc/ui/main_avtivity/MainPresenter.kt
package com.hfad.avc.ui.main_avtivity

import com.github.terrakok.cicerone.Router
import com.hfad.avc.cicerone.Screens.main
import com.hfad.avc.dagger.ComponentManager.Companion.instance
import moxy.InjectViewState
import moxy.MvpPresenter
import javax.inject.Inject

@InjectViewState
class MainPresenter : MvpPresenter<IMainViewModel>() {
   @Inject
   lateinit var router: Router

   init {
      instance.appComponent.inject(this)
   }

   override fun onFirstViewAttach() {
      super.onFirstViewAttach()
      router.navigateTo(main())
   }
}